src/hotspot/share/utilities/lineReader.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "memory/allocation.inline.hpp"
+#include "runtime/os.hpp"
+#include "utilities/lineReader.hpp"
+
+LineReader::LineReader(const char* filename) : _filename(filename), _stream(nullptr) {
+  // Use os::open() because neither fopen() nor os::fopen()
+  // can handle long path name on Windows. HMM, is this still valid today???
```

Comment:
It is not clear to me that any of these API's avoid the MAX_PATH 260 character limitation as the only way around that is to use `\\?\` UNC path naming - which we don't appear to do anywhere. On Windows 10 and later if  the machine/user is configured for long paths by default then the `FindFirstFile` used by `os::open` may avoid the 260 limit, but only if it actually maps to the underlying unicode version of the method.
Anyway something to follow up on in a separate RFE I think.